Cython/Utils.py

```diff
@@ -626,3 +626,23 @@ def build_hex_version(version_string):
         hexversion = (hexversion << 8) + digit
 
     return '0x%08X' % hexversion
+
+
+def write_depfile(mainfile, source, dependencies):
+    src_base_dir, _ = os.path.split(source)
```

Comment:
I'd also rename `mainfile` to `target_file` to distinguish it from `source`.
```suggestion
def write_depfile(mainfile, source_file, dependencies):
    src_base_dir = os.path.dirname(source_file)
```